accounts/abi/abigen/source2.go.tpl

```diff
@@ -0,0 +1,213 @@
+{{- /* -*- indent-tabs-mode: t -*- */ -}}
+// Code generated via abigen V2 - DO NOT EDIT.
+// This file is a generated binding and any manual changes will be lost.
+
+package {{.Package}}
+
+import (
+	"math/big"
+	"errors"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind/v2"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+// Reference imports to suppress errors if they are not otherwise used.
+var (
+	_ = errors.New
+	_ = big.NewInt
+	_ = common.Big1
+	_ = types.BloomLookup
+	_ = abi.ConvertType
+)
+
+{{$structs := .Structs}}
+{{range $structs}}
+	// {{.Name}} is an auto generated low-level Go binding around an user-defined struct.
+	type {{.Name}} struct {
+	{{range $field := .Fields}}
+	{{$field.Name}} {{$field.Type}}{{end}}
+	}
+{{end}}
+
+{{range $contract := .Contracts}}
+	// {{.Type}}MetaData contains all meta data concerning the {{.Type}} contract.
+	var {{.Type}}MetaData = bind.MetaData{
+		ABI: "{{.InputABI}}",
+		Pattern: "{{index $.Libraries .Type}}",
+		{{if .InputBin -}}
+		Bin: "0x{{.InputBin}}",
+		{{end -}}
+		{{if .Libraries -}}
+		Deps: []*bind.MetaData{
+		{{- range $name, $pattern := .Libraries}}
+			&{{$name}}MetaData,
+		{{- end}}
+		},
+		{{end}}
```

Comment:
```suggestion
		{{if .InputBin -}}
		Bin: "0x{{.InputBin}}",
		{{end -}}
		{{if .Libraries -}}
		Pattern: "{{index $.Libraries .Type}}",
		Deps: []*bind.MetaData{
		{{- range $name, $pattern := .Libraries}}
			&{{$name}}MetaData,
		{{- end}}
		},
		{{end}}
```

Reply:
If we aren't parsing the combined json, don't include the library pattern (the resulting bindings won't be used for deployment because they don't contain the deployer bytecode, so this just removes an unnecessary line that sets pattern to "")

Reply:
No this is incorrect:  If a contract has no libraries, it might be a library itself.  Unsure how to deal with this extra line of cruft in the generated bindings, so I've reverted it for now.